### DISCUSSION TITLE
Migrating to Chai 5.0 in browser

### DISCUSSION THREAD

**Discussion by schrotie** [chaijs/chai#1600] [Support] [answered]
This took me some time to figure out, a documentation update might help.

TLDR:
You need to change 

    <script src="../node_modules/chai/chai.js"></script>
to

    <script src="../node_modules/chai/chai.js" type="module"></script>

If you are using imperative Chai API calls like `chai.should()` you must use it in your own ESMs like this:

		import {should} from '../node_modules/chai/chai.js';
		should(); // was: chai.should()

The rest is documenting my journey to get it working again (I thought I couldn't fix this and wanted to file a bug for that):

Chai stopped working in the browser.

Old setup:

    <script src="../node_modules/chai/chai.js"></script>

This now fails with

    Uncaught SyntaxError: Unexpected token 'export' (at chai.js:3719:1)

Making it a module:

    <script src="../node_modules/chai/chai.js" type="module"></script>

Fails with

    Uncaught ReferenceError: chai is not defined

because it does no longer define the global "chai" (im using `chai.should()`, otherwise this *might* work)

Next try:

    import {chai} from '../node_modules/chai/chai.js';
    ->
    Uncaught SyntaxError: The requested module '../node_modules/chai/chai.js' does not provide an export named 'chai'

Aaaah, got it:

		import {should} from '../node_modules/chai/chai.js';

		should(); // was: chai.should()

**Comment by 43081j** [comment, marked as answer]
you are right, we need to update the docs (actually there's a few places we really should update asap before more people get confused)

basically, moving it to an ES module means we don't expose it on a global anymore automatically

so you would need to register one of the interfaces:

```ts
import '../node_modules/chai/register-should.js';
// or
import '../node_modules/chai/register-assert.js';
```

or alternatively do what you said, import chai within your JS (`import {chai}`)

**Comment by schrotie** [comment]
Okay, would a PR on https://github.com/chaijs/chaijs.github.io help?

**Reply by 43081j** [reply]
if you could, that'd be much appreciated. i haven't dealt with the website repo much so you might have some digging to do (to find where to make the change), but i'll ping the other maintainers to review once you have a pr up

**Comment by schrotie** [comment]
Ugh, sorry, I'd need a ruby system to run this which I don't have. Apologies, I'll not work on this.

**Reply by 43081j** [reply]
don't worry about it! could you make an issue in the repo still so we can track it?

one of us will try pick it up at some point soon im sure

**Comment by schrotie** [comment]
Done, find it [here](https://github.com/chaijs/chaijs.github.io/issues/205). 

I also _did_ some work after all but cannot verify it so I created a patch. See the linked comment.
